#! /bin/sh /usr/share/dpatch/dpatch-run
## 18_vdr-1.3.39-menu-on-control.dpatch by Udo Richter <udo_richter@gmx.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes main menu doesn't open if any replay control is active.

@DPATCH@
--- vdr-1.3.39-old/vdr.c	2006-01-15 18:33:54.000000000 +0100
+++ vdr-1.3.39/vdr.c	2006-01-15 18:38:28.000000000 +0100
@@ -795,8 +795,12 @@
                bool WasMenu = Interact && Interact->IsMenu();
                if (Menu)
                   DELETE_MENU;
-               else if (cControl::Control() && cOsd::IsOpen())
-                  cControl::Control()->Hide();
+               else if (cControl::Control()) {
+                  if (cOsd::IsOpen())
+                     cControl::Control()->Hide();
+                  else
+                     WasOpen = false;
+                  }
                if (!WasOpen || !WasMenu && !Setup.MenuButtonCloses)
                   Menu = new cMenuMain;
                }

